modules/platform/nuxeo-drive-server/nuxeo-drive-mongodb/src/main/java/org/nuxeo/drive/mongodb/MongoDBAuditChangeFinder.java

```diff
@@ -0,0 +1,195 @@
+            queryBuilder.and(Predicates.lt(LOG_LOG_DATE, lastClusteringInvalidationDate));
+        }
+        queryBuilder.order(OrderByExprs.desc(LOG_ID));
+        queryBuilder.limit(1);
+        List<LogEntry> entries = auditService.queryLogs(queryBuilder);
+
+        if (entries.isEmpty()) {
+            if (clusteringDelay > -1) {
+                // Check for existing entries without the clustering invalidation date filter to not return -1 in this
+                // case and make sure the lower bound of the next call to NuxeoDriveManager#getChangeSummary will be >=
+                // 0
+                if (auditService.getAuditCollection().estimatedDocumentCount() == 0) {
+                    log.debug("Found no audit log entries matching the criterias but some exist, returning 0");
+                    return 0;
+                }
+            }
+            log.debug("Found no audit log entries, returning -1");
+            return -1;
+        }
+        return entries.get(0).getId();
+    }
+
+    @Override
+    protected List<LogEntry> queryAuditEntries(CoreSession session, SynchronizationRoots activeRoots,
+            Set<String> collectionSyncRootMemberIds, long lowerBound, long upperBound, int limit) {
+        MongoDBAuditBackend auditService = (MongoDBAuditBackend) Framework.getService(AuditReader.class);
+        MongoCollection<Document> auditCollection = auditService.getAuditCollection();
+
+        // Build intermediate filters
+        Bson lifeCycleEvent = and(eq("category", "eventLifeCycleCategory"), eq("eventId", "lifecycle_transition_event"),
+                ne("docLifeCycle", "deleted"));
+        Bson documentEvent = and(eq("category", "eventDocumentCategory"),
+                in("eventId", "documentCreated", "documentModified", "documentMoved", "documentCreatedByCopy",
+                        "documentRestored", "addedToCollection", "documentProxyPublished", "documentLocked",
+                        "documentUnlocked", "documentUntrashed"));
+        Bson inRootPath = regex("docPath", getRootsRegex(activeRoots.getPaths()));
+        Bson inRootId = in("docUUID", collectionSyncRootMemberIds);
+        Bson sessionRepository = eq("repositoryId", session.getRepositoryName());
+        Bson driveCategory = eq("category", "NuxeoDrive");
+        Bson notRootUnregistered = ne("eventId", "rootUnregistered");
+        Bson isGeneral = and(or(documentEvent, lifeCycleEvent), or(inRootPath, inRootId));
+        Bson isDrive = and(driveCategory, notRootUnregistered);
+        Bson idRange = and(gt("_id", lowerBound), lte("_id", upperBound));
+
+        // Build final filter
+        Bson filter = and(sessionRepository, or(isGeneral,
+                isDrive), idRange);
+
+        BsonDocument bsonDocument = filter.toBsonDocument(Document.class, auditCollection.getCodecRegistry());
+        log.debug("Query on MongoDB-Audit: {}", bsonDocument.toJson());
```

Comment:
The query evaluation should be done in a lazy way:
```suggestion
        log.debug("Query on MongoDB-Audit: {}", () -> filter.toBsonDocument(Document.class, auditCollection.getCodecRegistry()).toJson());
```

Reply:
Totally forgot that aspect